src/ui/transform.js

```diff
@@ -0,0 +1,26 @@
+import moment from 'moment';
+import type { Message } from '../store/types';
+import { unescape } from '../utility/misc';
+
+// TODO don't mix validation and data munging
+export const messageToDetails = (email: Message) => {
+  const headers = email.payload.headers;
+
+  const fromHeader = headers.find((h) => h.name === 'From');
+  const subjectHeader = headers.find((h) => h.name === 'Subject');
+
+  if(!fromHeader) {
+    throw 'Email has no "From" header';
+  }
+  if(!subjectHeader) {
```

Comment:
Seems weird to throw an error if the email doesn't have a subject since that could happen frequently... sender I get :) Also, are you catching the exception anywhere?

Reply:
Mainly just trying to keep Flow happy here; is there a better way?  Not catching it anywhere (yet), but I may very well refactor this anyway, since I see "is the data valid" and "convert from format A to format B" as two separate concerns.

Reply:
I agree with Marie. Null subject is perfectly valid, IMO. Since you're doing all your data formatting logic in this here decorator function, it seems like just as good a place as any to set `subject = subjectHeader ? subjectHeader.value : 'No subject'`